test/e2e/vsphere/hostzonal.go

```diff
@@ -0,0 +1,317 @@
+package vsphere
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	vapirest "github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/utils/ptr"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:VSphereHostVMGroupZonal][platform:vsphere] Managed cluster should", func() {
+	defer GinkgoRecover()
+	ctx := context.Background()
+
+	var (
+		cfg *rest.Config
+		c   *kubernetes.Clientset
+		cc  *configclient.ConfigV1Client
+
+		vsphereCreds *corev1.Secret
+		err          error
+		infra        *configv1.Infrastructure
+		nodes        *corev1.NodeList
+	)
+
+	BeforeEach(func() {
+		cfg, err = e2e.LoadConfig()
+		Expect(err).NotTo(HaveOccurred())
+		c, err = e2e.LoadClientset()
+		Expect(err).NotTo(HaveOccurred())
+		cc, err = configclient.NewForConfig(cfg)
+		Expect(err).NotTo(HaveOccurred())
+		By("Get Infrastructure spec")
+		infra, err = cc.Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		if !isVmHostZonal(infra.Spec.PlatformSpec.VSphere) {
+			Skip("skipping test since cluster does not support vSphere host zones")
+		}
+
+		By("Get vSphere Credentials")
+		vsphereCreds, err = c.CoreV1().Secrets("kube-system").Get(ctx, "vsphere-creds", v1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Expect Failure Domains to be greater than one")
+		Expect(len(infra.Spec.PlatformSpec.VSphere.FailureDomains) >= 1)
+
+		nodes, err = c.CoreV1().Nodes().List(ctx, v1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	It("machine should be in the correct vm-host group", func() {
```

Comment:
I would add `[Suite:openshift/conformance/parallel]` to the end of these It() calls so that they are included in the parallel tests.